pkg/scheduler/controller/shoot/scheduler_control.go

```diff
@@ -193,10 +205,29 @@ func determineSeed(shoot *gardencorev1beta1.Shoot, seedLister gardencorelisters.
 	return getSeedWithLeastShootsDeployed(filteredCandidates, shootList)
 }
 
+func isUsableSeed(seed *gardencorev1beta1.Seed) bool {
+	return seed.DeletionTimestamp == nil && !gardencorev1beta1helper.TaintsHave(seed.Spec.Taints, gardencorev1beta1.SeedTaintInvisible) && common.VerifySeedReadiness(seed)
+}
+
+func filterUsableSeeds(seedList []*gardencorev1beta1.Seed) ([]*gardencorev1beta1.Seed, error) {
+	var matchingSeeds []*gardencorev1beta1.Seed
+
+	for _, seed := range seedList {
+		if isUsableSeed(seed) {
+			matchingSeeds = append(matchingSeeds, seed)
+		}
+	}
+
+	if len(matchingSeeds) == 0 {
+		return nil, fmt.Errorf("none of the %d seeds is valid for scheduling (not deleting, visible and ready)", len(seedList))
+	}
+	return matchingSeeds, nil
+}
+
 func filterSeedsMatchingSeedSelector(cloudProfile *gardencorev1beta1.CloudProfile, seedList []*gardencorev1beta1.Seed) ([]*gardencorev1beta1.Seed, error) {
 	selector := &metav1.LabelSelector{}
-	if cloudProfile.Spec.SeedSelector != nil {
-		selector = cloudProfile.Spec.SeedSelector
+	if cloudProfile.Spec.SeedSelector != nil && cloudProfile.Spec.SeedSelector.LabelSelector != nil {
```

Comment:
can we not exit here if cloudProfile.Spec.SeedSelector.LabelSelector == nil ?. If the CloudProfile does not specify a label selector, none of them are filtered out based on that criteria? 
This way we do not have to loop over the seeds.